pkg/operator/ceph/csi/csi.go

```diff
@@ -40,10 +41,22 @@ func ValidateAndConfigureDrivers(context *clusterd.Context, namespace, rookImage
 	}
 
 	if CSIEnabled() {
-		if err := startDrivers(context.Clientset, context.RookClientset, namespace, serverVersion, ownerInfo, v); err != nil {
-			logger.Errorf("failed to start Ceph csi drivers. %v", err)
-			return
+		csiLock.Lock()
+		for i := maxRetries - 1; i >= 0; i-- {
+			if err := startDrivers(context.Clientset, context.RookClientset, namespace, serverVersion, ownerInfo, v); err != nil && strings.Contains(err.Error(), "error reading ConfigMap") {
```

Comment:
Why searching for this error only? Why not any error?

Reply:
@leseb , The main goal of issue #7950 was to address k8s error related to configmap which prevented csi driver start.  
```
2021-05-19 10:32:06.905743 I | ceph-csi: Kubernetes version is 1.20
2021-05-19 10:32:07.595903 E | ceph-csi: failed to start Ceph csi drivers. failed to load ROOK_CSI_RESIZER_IMAGE setting: error reading ConfigMap "rook-ceph-operator-config". etcdserver: leader changed
```
The only place such an error may occur while reading csi parameters was at 
https://github.com/rook/rook/blob/4bdc02e16e26b07d1f232c95e748f248740e36c3/pkg/operator/k8sutil/configmap.go#L60 

Regarding handling any error that comes up, this may lead to unnecessary looping in case of bad/ill-formatted option presented through env var/ configmap.
So, it's kind of trade off in between them. I can remove that check and loop over for any error if that's favoured.

Reply:
@Rakshith-R  I think we should be checking for err which is coming like `etcdserver: leader changed` and seems like retry creating csi-driver will fix #7483 @leseb ?

Reply:
@subhamkrai IMO, it is not entirely feasible to list down kubernetes errors and separate them into categories which we deem needs a retry and which does not. If you have any particular suggestions, I'd be happy to look into it. 

The main places where k8s error may occur are mostly during configmap reads ( #7950) and creating csidriver resources(#7483), between either of them configmap reads are more frequent. 

We can either retry on all errors while risking retries for ill-formatted/wrong options or retry for errors containing a close match to what we deem needs a retry(like `error reading ConfigMap`).
>So, it's kind of trade off in between them. I can remove that check and loop over for any error if that's preferred.

Reply:
right, looks like we'll have to go with `error reading ConfigMap` if we don't want to retry in all errors.

Reply:
Let's just retry on any failure. Failures are expected to be rare, and we're only retrying three times anyway.